py/htm/encoders/simhash_document_encoder.py

```diff
@@ -0,0 +1,158 @@
+# under the terms of the GNU Affero Public License version 3 as published by
+# the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero Public License for
+# more details.
+#
+# You should have received a copy of the GNU Affero Public License along with
+# this program.  If not, see http://www.gnu.org/licenses.
+# ------------------------------------------------------------------------------
+
+import htm.bindings.encoders
+SimHashDocumentEncoder = htm.bindings.encoders.SimHashDocumentEncoder
+SimHashDocumentEncoderParameters = htm.bindings.encoders.SimHashDocumentEncoderParameters
+
+__all__ = ['SimHashDocumentEncoder', 'SimHashDocumentEncoderParameters']
+
+
+if __name__ == '__main__':
+  """
+  Simple program to examine the SimHashDocumentEncoder.
+
+  For help using this program run:
+  $ python -m htm.encoders.simhash_document_encoder --help
+  """
+  import argparse
+  import numpy
+  import random
+  import sys
+  import textwrap
+
+  from sys import exit
+
+  from htm.bindings.sdr import SDR, Metrics
+
+
+  # Gather input from the user.
+  parser = argparse.ArgumentParser(
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+    description="Simple program to examine the SimHashDocumentEncoder.\n\n" +
+    textwrap.dedent(SimHashDocumentEncoder.__doc__ + "\n\n" +
+      SimHashDocumentEncoderParameters.__doc__))
+  parser.add_argument('--activeBits', type=int, default=0,
+    help=SimHashDocumentEncoderParameters.activeBits.__doc__)
+  parser.add_argument('--caseSensitivity', action='store_true', default=False,
+    help=SimHashDocumentEncoderParameters.caseSensitivity.__doc__)
+  parser.add_argument('--size', type=int, default=0,
+    help=SimHashDocumentEncoderParameters.size.__doc__)
+  parser.add_argument('--sparsity', type=float, default=0,
+    help=SimHashDocumentEncoderParameters.sparsity.__doc__)
+  parser.add_argument('--tokenSimilarity', action='store_true', default=False,
+    help=SimHashDocumentEncoderParameters.tokenSimilarity.__doc__)
+  args = parser.parse_args()
+
+  # Copy the command line arguments into the parameter structure.
+  parameters = SimHashDocumentEncoderParameters()
+  parameters.activeBits = args.activeBits
+  parameters.caseSensitivity = args.caseSensitivity
+  parameters.size = args.size
+  parameters.sparsity = args.sparsity
+  parameters.tokenSimilarity = args.tokenSimilarity
+
+  # Try initializing the encoder.
+  try:
+    encoder = SimHashDocumentEncoder(parameters)
+  except RuntimeError as error:
+    print(error)
+    parser.print_usage()
+    exit()
+
+  # Run the encoder and measure some statistics about its output.
+  num_samples = 1000 # number of documents to run
+  num_tokens = 10 # tokens per document
+  testCorpus = [ "find", "any", "new", "work", "part", "take", "get", "place",
```

Comment:
this is a good example, but seems a lot of work/code that is unrelated to the encoder itself. Consider separating into:
- an example in `py/htm/examples/sim_hash_document_encoder_example.py`? 
- or as a part of the py unit tests, we have sometimes "exampleUsageTest()"
What do you think?